Inferences/BinaryResolution.cpp

```diff
@@ -113,9 +113,16 @@ Clause* BinaryResolution::generateClause(Clause* queryCl, Literal* queryLit, Cla
   Inference inf(GeneratingInference2(nConstraints == 0 ?  InferenceRule::RESOLUTION : InferenceRule::CONSTRAINED_RESOLUTION, queryCl, resultCl));
   Inference::Destroyer inf_destroyer(inf); // will call destroy on inf when coming out of scope unless disabled
 
-  bool needsToFulfilWeightLimit = passiveClauseContainer && !passiveClauseContainer->fulfilsAgeLimit(wlb, numPositiveLiteralsLowerBound, inf) && passiveClauseContainer->weightLimited();
+  bool andThatsIt = false;
+  bool hasAgeLimitStrike = passiveClauseContainer && passiveClauseContainer->mayBeAbleToDiscriminateClausesUnderConstructionOnLimits()
+                                                  && passiveClauseContainer->exceedsAgeLimit(numPositiveLiteralsLowerBound, inf, andThatsIt);
+  if (hasAgeLimitStrike && andThatsIt) { // we are dealing with purely age-limited container (no need for weight-related investigations)
```

Comment:
What are the semantics of `andThatsIt`? The container only discriminates by age?

Reply:
Yes, it's the weird `-arw 1:0` corner case. 

I agree we could simplify things here if we agreed that pure age container is not worth extra fancyness in the code.

Otherwise, I was just trying to do my homework to the end ;)

Reply:
'A' for the homework! OK - I was just thinking if so that you could have a member function `isPureAgeContainer() const` or even a static (if the inheritance stuff works out) that does that bit, and thereby remove the need for the out-parameter here.

Reply:
It's definitely to the effect of `isPureAgeContainer`. But I think we would need another virtual function in the interface for this, don't you?

I guess I can still easily kick this out, check how much we lose in the `-awr 1:0` case and if its bearable, simplify, but leave a comment on how the homework should have been done?

Reply:
Please don't kick it out, it seems like the nice way of doing things! I intended adding another virtual function to the interface like you say, but if you don't like that solution this one seems fine to me. :-)